src/edit/index.vue

```diff
@@ -1,9 +1,116 @@
 <template>
-  <p>Custom element template</p>
+  <div class="tce-api-video">
+    <element-placeholder
+      v-if="isEmpty"
+      :is-focused="isFocused"
+      :is-disabled="isDisabled"
+      name="Api Video component"
+      icon="mdi-video-image"
+      active-placeholder="Use toolbar to upload the video"
+      active-icon="mdi-arrow-up" />
+    <div v-else>
+      <tce-overlay v-if="error || didUploadFail" type="error">
+        <v-icon color="error">mdi-alert</v-icon>
+        {{ error || 'Video upload failed. Please try again.' }}
+      </tce-overlay>
+      <tce-overlay v-if="infoMessage">
+        <v-progress-circular indeterminate color="info" class="mr-4" />
+        {{ infoMessage }}
+      </tce-overlay>
+      <div ref="player" class="player"></div>
+    </div>
+  </div>
 </template>
 
 <script>
+import createUpload from '../createUpload';
+import { ELEMENT_STATE } from '../shared';
+import ElementPlaceholder from '../tce-core/ElementPlaceholder.vue';
+import get from 'lodash/get';
+import TceOverlay from './Overlay.vue';
+
+const DEFAULT_ERROR_MSG = 'Something went wrong.';
+const UPLOADING_MSG = 'Video is uploading... Do not leave the page.';
+const PROCESSING_MSG = 'Video is processing...';
+
 export default {
-  name: 'tce-api-video'
+  name: 'tce-api-video',
+  inject: ['$elementBus'],
+  props: {
+    element: { type: Object, required: true },
+    isFocused: { type: Boolean, default: false },
+    isDisabled: { type: Boolean, default: false }
+  },
+  data: () => ({ file: null }),
+  computed: {
+    videoId: ({ element }) => element.data?.videoId,
+    uploadUrl: ({ element }) => element.data?.uploadUrl,
+    embedCode: ({ element }) => element.data?.embedCode,
+    fileName: ({ element }) => element.data?.fileName,
+    playable: ({ element }) => element.data?.playable,
+    status: ({ element }) => element.data?.status,
+    error: ({ element }) => element.data?.error,
```

Comment:
Most of these are not used in many places. Do you think it would be better / reduce noise if we'd remove them and read data straight from the `element` instead of flattening the structure?